floris/floris_model.py

```diff
@@ -210,6 +210,22 @@ def _reinitialize(
         if air_density is not None:
             flow_field_dict["air_density"] = air_density
         if heterogeneous_inflow_config is not None:
+
+            # If both heterogeneous_inflow_config contains a z term,
+            # and flow_field_dict["wind_shear"] is
+            # not 0, raise and error the user that there is potential
+            # for double-counting of shear
+            if (
+                "z" in heterogeneous_inflow_config
+                and flow_field_dict["wind_shear"] != 0.0
+            ):
+                if heterogeneous_inflow_config['z'] is not None:
+                    raise ValueError(
```

Comment:
We could consider using logger warning here, rather than a ValueError, in case the user would like to have both wind shear and a z component. It's probably an edge case, but I could imagine there might be times when it could be useful.

Reply:
I started with a warning, but I couldn't think of a case where you'd want both, also I think the result is a little ambiguous, for example, say you apply a normal shear coefficient, but then use heterogeneity to have the shear inverted and go slower with height.  You will get some result, but then you don't have the shear layer implied by the exponent and that feels more like an error to me.

 It was really easy to imagine accidentally forgetting to turn off shear (I had done it!)

Reply:
Yeah, definitely easy to forget, but that's what the warning would be for right? Or is your feeling that the warnings are too easy to ignore?

Reply:
I guess it's more I'm wondering if it isn't an error, what I mean is, the shear layer will be a little unpredictable to me if both are applied.  I think of it like applying yaw and derating at the same time, it could work but we don't allow it because the correct result is unclear

Reply:
Oh, I see. Yeah, Ok, that makes sense. Maybe we should just add to the error statement something like: "please include shear effects in `speed_multipliers`". At the moment, the wording of the error sounds like a warning ("may result in double-counting"). Since it won't run, there's no risk of it double-counting.